grafana/adapters.go

```diff
@@ -39,3 +51,14 @@ func int32SliceToIntList(list []int32) []interface{} {
 func int32SliceToSet(src []int32) *schema.Set {
 	return schema.NewSet(schema.HashInt, int32SliceToIntList(src))
 }
+
+func listOfSetsToStringSlice(listSet []interface{}) [][]string {
+	var ret [][]string
+	if listSet == nil {
+		return ret
+	}
```

Comment:
There's no need for this `if`: calling `range` on a `nil` slice is the same as calling it over an empty slice. What I suggest doing instead is allocating `ret` with the required size:
```suggestion
	ret := make([][]string, 0, len(listSet))
```